modules/auxiliary/gather/censys_search.rb

```diff
@@ -46,14 +46,13 @@ def search(keyword, search_type)
         'query' => keyword
       }
 
-      @cli = Rex::Proto::Http::Client.new('www.censys.io', 443, {}, true)
+      @cli = Rex::Proto::Http::Client.new('search.censys.io', 443, {}, true)
       @cli.connect
 
       response = @cli.request_cgi(
         'method' => 'post',
-        'uri' => "/api/v1/search/#{search_type}",
+        'uri' => "/api/v2/hosts/search/#{keyword}",
```

Comment:
My concern here is that this will break on v1.  I know v2 is apparently being deployed as a rolling release.  Is there a way for us to use v1, and if it fails, use v2?

Reply:
The v1 is disabled for host search: it does not return 200.
If it was in the midst of change and still available we could use v1, check the response, and in case of blank use v2, yet this is vain now.

Reply:
I got an error when testing this module:
```
msf6 auxiliary(gather/censys_search) > run CENSYS_DORK=rapid7 CENSYS_UID=<redacted> CENSYS_SECRET=<redacted>

[-] Auxiliary failed: JSON::ParserError 859: unexpected token at '<html>
<head><title>400 Bad Request</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<hr><center>cloudflare</center>
</body>
</html>
'
[-] Call stack:
[-]   /home/msfuser/.rvm/gems/ruby-3.0.2@metasploit-framework/gems/json-2.6.2/lib/json/common.rb:216:in `parse'
[-]   /home/msfuser/.rvm/gems/ruby-3.0.2@metasploit-framework/gems/json-2.6.2/lib/json/common.rb:216:in `parse'
[-]   /home/msfuser/.rvm/gems/ruby-3.0.2@metasploit-framework/gems/activesupport-6.1.6/lib/active_support/json/decoding.rb:23:in `decode'
[-]   /home/msfuser/metasploit-framework/modules/auxiliary/gather/censys_search.rb:69:in `search'
[-]   /home/msfuser/metasploit-framework/modules/auxiliary/gather/censys_search.rb:166:in `run'
[*] Auxiliary module execution completed
```

It looks like the `cloud_lookup` module sends a `GET` request with a `q=` parameter?
```ruby
        'method' => 'GET',
        'uri' => "/api/v2/hosts/search?q=#{keyword}",
```

Reply:
Also, it looks like other fixes are needed to make this module work. The JSON returned by this API seems to have changed. For example, line 70:
```
results = records['results']
```
This should be:
```
results = records['result']
```